site/content/documentation/programming/shacl/index.md

```diff
@@ -14,125 +14,300 @@ This documentation is for RDF4J 2.5 and onwards, as the SHACL engine was experim
 
 # How does the SHACL engine work
 
-The SHACL engine works by analyzing the changes made in a transaction and creating a set of validation plans (similar to query plans) and executing these as part of the transaction `commit()` call.
+The SHACL engine works by analyzing the changes made in a transaction and creating a set of validation 
+plans (similar to query plans) and executing these as part of the transaction `commit()` call.
 
-In many cases the SHACL engine can validate your changes based on your changes alone. In some cases your changes may affect or be affected by data already in the database, in which case the engine will query the database for that particular data.
+In many cases the SHACL engine can validate your changes based on your changes alone. 
+In some cases your changes may affect or be affected by data already in the database, in which case the 
+engine will query the database for that particular data.
 
 Here is an example of when data in the database affects the validation plans:
 
-    ex:PersonShape
-            a sh:NodeShape  ;
-            sh:targetClass ex:Person ;
-            sh:property [
-                    sh:path ex:age ;
-                    sh:datatype xsd:integer ;
-            ] .
+```turtle
+ex:PersonShape
+    a sh:NodeShape  ;
+    sh:targetClass ex:Person ;
+    sh:property [
+        sh:path ex:age ;
+        sh:datatype xsd:integer ;
+    ] .
+```
 
 Initial data in the database.
 
-    ex:pete a ex:Person.
-
+```
+ex:pete a ex:Person.
+```
 Data added by a transaction.
 
-    ex:pete ex:age "eighteen".
+```
+ex:pete ex:age "eighteen".
+```
+
+For this example the SHACL engine will match the predicate `ex:age` with `ex:PersonShape` and realise that `ex:pete` 
+might already be defined as a `ex:Person` in the database. The validation plan will then include checking if `ex:pete` 
+is type `ex:Person`.
 
-For this example the SHACL engine will match the predicate `ex:age` with `ex:PersonShape` and realise that `ex:pete` might already be defined as a `ex:Person` in the database. The validation plan will then include checking if `ex:pete` is type `ex:Person`.
+# How to load and update SHACL shapes
 
-# How to load SHACL shapes
+The ShaclSail uses a reserved graph (`http://rdf4j.org/schema/rdf4j#SHACLShapeGraph`) for storing the SHACL shapes. 
+Utilize a normal connection to load your shapes into this graph. SPARQL is not supported.
 
-The ShaclSail uses a reserved graph (`http://rdf4j.org/schema/rdf4j#SHACLShapeGraph`) for storing the SHACL shapes. Utilize a normal connection to load your shapes into this graph. 
+```java
```

Comment:
This doesn't work as expected. It formats as a preformatted code block but uses no syntax highlighting. You need to use the `{{< highlight java >}}` shortcode instead for that.

Reply:
What processor are we using for our markdown? The {{}} syntax isn’t recognized by any or the ones I am familiar with.

Reply:
We use Hugo to generate the docs on the website (https://gohugo.io/templates/introduction/)

Reply:
It looks like it should be working in 0.60.0 https://gohugo.io/content-management/syntax-highlighting/#highlighting-in-code-fences

Reply:
I can have a look at upgrading our hugo instance. Logged as https://github.com/eclipse/rdf4j/issues/1902

Reply:
Unfortunately hugo 0.60 has several non-compatible changes, I want to take a good look at those before upgrading.

There's also the fact that we use the hugo shortcode everywhere else. The triple backtick block gets rendered different from our other code blocks. I'll push a fix to this PR to use the old syntax for now so we can merge this, and then we can start looking at upgrading to a newer hugo version as a separate issue.